packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1089,16 +1104,68 @@ export function getSdkModelPropertyType(
       })
     );
   }
-  return diagnostics.wrap({
+  const multiPartOption: MultipartOptionsType = {
+    isNameDefined: httpOperationPart ? httpOperationPart.name !== undefined : true,
+    isFilePart: isBytesInput && operationIsMultipart,
+    multi: httpOperationPart ? httpOperationPart.multi : isRawBytesArray,
+    headers: [], // TODO
+    filename: httpOperationPart?.filename
+      ? diagnostics.pipe(getSdkModelPropertyType(context, httpOperationPart.filename, operation))
+      : undefined,
+    contentType: httpOperationPart?.body.contentTypeProperty
+      ? diagnostics.pipe(
+          getSdkModelPropertyType(context, httpOperationPart.body.contentTypeProperty, operation)
+        )
+      : undefined,
+  };
+  base.isMultipartFileInput = multiPartOption.isFilePart;
+  base.multipartOptions = operationIsMultipart ? multiPartOption : undefined;
+  if (operationIsMultipart && base.type.kind === "array") {
+    if (httpOperationPart === undefined) {
+      // for "images: bytes[]", return type shall be "bytes" instead of "bytes[]"" and "multipartOptions.multi" shall be true
+      if (base.type.valueType.kind === "bytes") {
+        base.type = base.type.valueType;
+      }
+    } else if (httpOperationPart.multi) {
+      // for "images: HttpPart<T>[]", return type shall be "T" instead of "T[]"" and "multipartOptions.multi" shall be true
+      base.type = base.type.valueType;
+    }
+  }
+  return diagnostics.wrap(undefined);
+}
+
+export function getSdkModelPropertyType(
+  context: TCGCContext,
+  type: ModelProperty,
+  operation?: Operation,
+  httpOperationPart?: HttpOperationPart
+): [SdkModelPropertyType, readonly Diagnostic[]] {
```

Comment:
Mostly for @tadelesh  Why is `getSdkModelPropertyType` export? Is there other lib that use this API.

Ask because here its signature is changed, which means a breaks if any lib uses it.

Reply:
I understand that this function shall not be exposed.

Reply:
It is already used by some language emitters. The new added signature is optional and doesn't change old logic.